src/Compilers/Core/MSBuildTask/BuildClient.cs

```diff
@@ -31,11 +31,17 @@ internal static class BuildClient
         private const int TimeOutMsNewProcess = 20000; 
 
         /// <summary>
-        /// Try to get the directory this assembly is in.
+        /// Try to get the directory this assembly is in. Returns null if assembly
+        /// was in the GAC.
         /// </summary>
-        private static string GetClientDir()
+        private static string TryGetClientDir()
         {
-            var uri = new Uri(Assembly.GetExecutingAssembly().CodeBase);
+            var assembly = Assembly.GetExecutingAssembly();
```

Comment:
I'm not sure -- does GetEntryAssembly return the GAC'd assembly?

Also, it will return the location of the shadow copied task if one exists. Is that what we want?